pkg/kubelet/cm/cgroup_manager_linux.go

```diff
@@ -44,60 +161,93 @@ type cgroupManagerImpl struct {
 	// subsystems holds information about all the
 	// mounted cgroup subsytems on the node
 	subsystems *CgroupSubsystems
+	// simplifies interaction with libcontainer and its cgroup managers
+	adapter *libcontainerAdapter
 }
 
 // Make sure that cgroupManagerImpl implements the CgroupManager interface
 var _ CgroupManager = &cgroupManagerImpl{}
 
 // NewCgroupManager is a factory method that returns a CgroupManager
-func NewCgroupManager(cs *CgroupSubsystems) CgroupManager {
+func NewCgroupManager(cs *CgroupSubsystems, cgroupDriver string) CgroupManager {
+	managerType := libcontainerCgroupfs
+	if cgroupDriver == string(libcontainerSystemd) {
```

Comment:
i am going to keep the input argument as is.  the idea is that this interface keeps you from needing to expose libcontainer concepts too much.  we can revisit it later if it proves problematic.  basically, i do not want to expose libcontainerCgroupAdapterType all the way up to cm.NodeConfig.CgroupDriver, so a conversion is needed somewhere.